### PR TITLE
Fix as_source_array/as_range_array for BlockRowOperator/BlockColumnOperator

### LINKED ISSUE #624: BlockRowOperator.as_source_array does not work
The error appears [here](https://github.com/pymor/pymor/blob/6fae9512109a4cdc079d40cda47207326136b4fb/src/pymor/operators/block.py#L154), because `BlockRowOperator.source` does not necessarily have the `subspaces` attribute.

Similar issue is in `BlockColumnOperator.as_range_array`.

### PR DESCRIPTION
**Pull request by sdrave**
Fixes #624.

### CODE REVIEW

File: src/pymor/operators/block.py [hunk 1]
```diff
@@ -139,7 +139,8 @@ def process_row(row, space):
                     R.append(op.as_range_array(mu))
             return R
 
-        blocks = [process_row(row, space) for row, space in zip(self._blocks, self.range.subspaces)]
+        subspaces = self.range.subspaces if self.blocked_range else self.range
```
Comment (pmli) [inline src/pymor/operators/block.py]: `subspaces` should be a sequence of vector spaces, right? So, maybe
```suggestion
        subspaces = self.range.subspaces if self.blocked_range else (self.range,)
```
instead?

Comment (sdrave) [reply src/pymor/operators/block.py]: Right, this is what I inteded to write ...

**Comment by sdrave** [comment]
@pmli, could you verify that the code is now working. (We don't have a test for this ATM ..)

**Comment by pmli** [comment]
It seems to be working
```python
>>> from pymor.basic import *
>>> from pymor.operators.block import BlockRowOperator
>>> import numpy as np
>>> A = NumpyMatrixOperator(np.eye(2))
>>> B = BlockRowOperator([A, A])
>>> B.as_source_array()
<pymor.vectorarrays.block.BlockVectorArray object at 0x7fa1a195a9e8>
>>> B.as_range_array()
NumpyVectorArray([[1. 0.]
 [0. 1.]
 [1. 0.]
 [0. 1.]], NumpyVectorSpace(2))
```

**Pull request merged**